pkg/epp/datastore/datastore_test.go

```diff
@@ -340,3 +343,121 @@ func TestMetrics(t *testing.T) {
 		})
 	}
 }
+
+func TestPods(t *testing.T) {
+	poolSelector := map[string]string{"app": "vllm_v1"}
+	pool := testutil.MakeInferencePool("pool").
+		Namespace("default").
+		Selector(poolSelector).ObjRef()
+	updatedPod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "pod1",
+		},
+		Spec: corev1.PodSpec{
+			NodeName: "node-1",
+		},
+	}
+	notReadyPod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "pod2",
+		},
+		Status: corev1.PodStatus{
+			Conditions: []corev1.PodCondition{{Type: corev1.PodReady, Status: corev1.ConditionFalse}},
+		},
+	}
+	tests := []struct {
+		name         string
+		op           func(ctx context.Context, ds Datastore)
+		existingPods []*corev1.Pod
+		wantPods     []*corev1.Pod
+	}{
+		{
+			name:         "Add new pod, no existing pods, should add",
+			existingPods: []*corev1.Pod{},
+			wantPods:     []*corev1.Pod{pod1},
+			op: func(ctx context.Context, ds Datastore) {
+				ds.PodUpdateOrAddIfNotExist(pod1, pool)
+			},
+		},
+		{
+			name:         "Add new pod, with existing pods, should add",
+			existingPods: []*corev1.Pod{pod1},
+			wantPods:     []*corev1.Pod{pod1, pod2},
+			op: func(ctx context.Context, ds Datastore) {
+				ds.PodUpdateOrAddIfNotExist(pod2, pool)
+			},
+		},
+		{
+			name:         "Update existing pod, new field, should update",
+			existingPods: []*corev1.Pod{pod1},
+			wantPods:     []*corev1.Pod{updatedPod},
+			op: func(ctx context.Context, ds Datastore) {
+				ds.PodUpdateOrAddIfNotExist(updatedPod, pool)
+			},
+		},
+		{
+			name:         "Update existing pod, no new fields, should not update",
+			existingPods: []*corev1.Pod{pod1},
+			wantPods:     []*corev1.Pod{pod1},
+			op: func(ctx context.Context, ds Datastore) {
+				incoming := &corev1.Pod{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "pod1",
+						Namespace: "default",
+					},
+				}
+				ds.PodUpdateOrAddIfNotExist(incoming, pool)
+			},
+		},
+		{
+			name:         "Add not ready pod, resync required, should update",
+			existingPods: []*corev1.Pod{pod1, notReadyPod},
+			wantPods:     []*corev1.Pod{pod1, pod2},
+			op: func(ctx context.Context, ds Datastore) {
+				scheme := runtime.NewScheme()
+				cfg := config.GetConfigOrDie()
+				cli, err := client.New(cfg, client.Options{Scheme: scheme})
+				if err != nil {
+					t.Fatalf("Unable to create ctrl runtime client")
+				}
+				ds.PodResyncAll(ctx, cli, pool)
+			},
```

Comment:
pod resync is not required on pod not ready. it is required when inf pool label selector is changed.
generally, I've been working on datastore a bit (started refactoring, pushed a small PR and more to come), 
IMO PodResyncAll should be removed from the interface.
having said that, the internal behavior is still there and is called upon a change in the pool label selector.
more details in the PR #713